### PR TITLE
 JBPM-8541: Process instance variable sorting is not working 

### PR DESCRIPTION
**Pull request by nmirasch**

### CODE REVIEW

**Comment by nmirasch** [comment]
just to advance the your review I have included  https://github.com/kiegroup/jbpm-wb/pull/1357 commit here, Please could you take a look on this process instance variables sorting fix? Thanks!

**Comment by nmirasch** [comment]
@cristianonicolai @jstastny-cz ..just to advance the your review I have included #1357 commit here, Please could you take a look on this process instance variables sorting fix? Thanks!

**Comment by cristianonicolai** [comment]
Jenkins execute full downstream build

**Comment by nmirasch** [comment]
rebased!

**Comment by nmirasch** [comment]
Jenkins execute full downstream build

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Comment by nmirasch** [comment]
rebased

**Comment by nmirasch** [comment]
ok to test

**Comment by cristianonicolai** [comment]
@nmirasch needs rebase now :)

**Comment by nmirasch** [comment]
@cristianonicolai done :)

**Comment by cristianonicolai** [comment]
Jenkins execute full downstream build

**Comment by nmirasch** [comment]
@jstastny-cz  There is one error in the full downstream but it seems not related, could you review?

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/main/java/org/jbpm/workbench/pr/backend/server/util/VariableHelper.java [hunk 1]
```diff
@@ -81,9 +85,27 @@ public static void registerProcessor(VariableProcessor processor) {
...
+                comparator = comparing(ProcessVariableSummary::getType);
+                break;
+            case COL_PROCESS_INSTANCE_VAR_LASTMOD:
+                comparator = comparing(ProcessVariableSummary::getTimestamp);
+                break;
+            default:
+                comparator = comparing(ProcessVariableSummary::getName);
+        }
+
+        variablesSummary.sort(!asc ? comparator : comparator.reversed());
```
Comment (jstastny-cz) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/main/java/org/jbpm/workbench/pr/backend/server/util/VariableHelper.java]: Isn't the logic reversed here? I'd assume default order for Comparator<T> is ascending (relying on compareTo() implementation of the types passed to comparing()) .... so when asc is true, I'd expect sorting using the comparator, when false by its reversed counterpart.

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/test/java/org/jbpm/workbench/pr/backend/server/util/VariableHelperTest.java [hunk 2]
```diff
@@ -91,4 +92,23 @@ public void addDeploymentIdServerTemplateToAllProcessInstancesVarsTest() {
...
         assertEquals("", varList.get(3).getNewValue());
     }
+
+    @Test
+    public void sortProcessInstancesVarsByNameTest() {
+        List<ProcessVariableSummary> varList =
+                VariableHelper.adaptCollection(variables, properties, processInstanceId, deploymentId, serverTemplateId,
+                                               COL_PROCESS_INSTANCE_VAR_ID, false);
+        assertEquals(4, varList.size());
+        assertEquals(processDefaultVar_name, varList.get(0).getName());
```
Comment (jstastny-cz) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/test/java/org/jbpm/workbench/pr/backend/server/util/VariableHelperTest.java]: The sort order seems to be really reversed .... the variable is called asc, and when false, we're happy with values being in ... ascending order.

**Comment by nmirasch** [comment]
@jstastny-cz updated

**Comment by jstastny-cz** [comment]
Jenkins execute full downstream build

**Comment by cristianonicolai** [comment]
Jenkins execute full downstream build

**Comment by nmirasch** [comment]
Jenkins execute full downstream build

**Comment by cristianonicolai** [comment]
@jstastny-cz build is green, happy for this to be merged?

**Comment by jstastny-cz** [comment]
@cristianonicolai not yet, I need to start the workbench from the downstream build and check. I should have it by the end of the day.

**Comment by nmirasch** [comment]
@jstastny-cz this fix is related with the table displayed at  the 'Process Variables' tab when a process instance is selected, here:  
procVar

Could you check it again? Thanks!

**Pull request merged**